src/hdnode.js

```diff
@@ -206,6 +206,15 @@ HDNode.prototype.toBase58 = function (__isPrivate) {
 
 // https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#child-key-derivation-ckd-functions
 HDNode.prototype.derive = function (index) {
+  typeforce('Number', index)
+  var highestIndex = (HDNode.HIGHEST_BIT * 2) // should be 0x100000000
+  if (index < 0) {
+    throw new Error('Could not derive with negative index')
+  }
+  if (index >= highestIndex) {
+    throw new Error('Could not derive with too high index')
+  }
+
```

Comment:
I have to say I don't understand typeforce that much, but it seems it's actually pretty easy to do that in there. How do you want to name the types... UnsignedInt31 and UnsignedInt32? :) (can typeforce right now to check if the number is integer by the way?)